### PR TITLE
add Accessor::global_index()

### PR DESCRIPTION
**Pull request by davydden**
fixes https://github.com/dealii/dealii/issues/6581

### CODE REVIEW

File: include/deal.II/lac/sparsity_pattern.h [hunk 1]
```diff
@@ -149,6 +149,13 @@ namespace SparsityPatternIterators
      */
     size_type index () const;
 
+    /**
+     * Return global index in sparsity pattern.
```
Comment (bangerth) [inline include/deal.II/lac/sparsity_pattern.h]: Say something like "This function returns the how-many'th entry within the entire sparsity pattern the current iterator points to. While the order of entries in a sparsity pattern is generally not important, this function allows indexing entries of the sparsity pattern and associated matrices using a linear index."

**Comment by bangerth** [comment]
Oh, also: no test?

**Comment by davydden** [comment]
> It's not entirely clear to me what this function can be useful for

useful when you use SparsityPatter outside of SparseMatrix class. I will add a test shortly...

**Comment by davydden** [comment]
@bangerth i added the test.

**Pull request merged**